### PR TITLE
Wrote tests for compressor functions

### PR DESCRIPTION
**Pull request by Azrenbeth**
BUILDS ON azren/level_tests (PR #47)

wrote tests for compressor functions

### CODE REVIEW

File: src/compressor.rs [hunk 1]
```diff
@@ -332,50 +332,440 @@ mod level_tests {
...
+        .collect();
+
+        for sg in 0i64..=13i64 {
+            assert_eq!(
+                expected_edges.get(&sg).cloned(),
+                new_state[&sg].prev_state_group,
+                "state group {} did not match expected",
+                sg,
+            );
+        }
```
Comment (erikjohnston) [inline src/compressor.rs:770]: This file is now getting quite long, I think maybe we should move such tests out into the `tests/` folder?

Comment (Azrenbeth) [reply src/compressor.rs:770]: The trouble with moving them into `tests/` is that I then can only use the things that are public - it is getting a bit long though!

Comment (Azrenbeth) [reply src/compressor.rs:770]: I could put them in seperate files by moving them into a `compressor/` folder:

So `src/compressor.rs` would have:
```rust
#[cfg(test)]
mod compressor_tests;
```

And then there would be a file `src/compressor/compressor_tests` that had
```rust
use xyz;

#[test]
fn test_this_does_that() {
    ...
}
````

would you be happy with that?

Comment (erikjohnston) [reply src/compressor.rs]: wfm!

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**